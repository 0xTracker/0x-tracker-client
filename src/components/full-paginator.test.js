import { cleanup, fireEvent, render } from 'react-testing-library';
import React from 'react';

import FullPaginator from './full-paginator';

const simpleProps = {
  page: 1,
  pageCount: 10,
  pageSize: 15,
  recordCount: 147,
};

afterEach(() => {
  cleanup();
});

it('should render full size paginator', () => {
  const { container } = render(<FullPaginator {...simpleProps} />);

  expect(container.firstChild).toMatchSnapshot();
});

it('should call onPageChange prop when going to specific page', () => {
  const handlePageChange = jest.fn();
  const { getByText } = render(
    <FullPaginator {...simpleProps} onPageChange={handlePageChange} />,
  );
  const link = getByText('3');

  fireEvent.click(link);

  expect(handlePageChange).toHaveBeenCalledTimes(1);
  expect(handlePageChange).toHaveBeenCalledWith(3);
});

it('should call onPageChange prop when going to previous page', () => {
  const handlePageChange = jest.fn();
  const { getByText } = render(
    <FullPaginator {...simpleProps} onPageChange={handlePageChange} page={3} />,
  );
  const link = getByText('Previous');

  fireEvent.click(link);

  expect(handlePageChange).toHaveBeenCalledTimes(1);
  expect(handlePageChange).toHaveBeenCalledWith(2);
});

it('should call onPageChange prop when going to next page', () => {
  const handlePageChange = jest.fn();
  const { getByText } = render(
    <FullPaginator {...simpleProps} onPageChange={handlePageChange} page={3} />,
  );
  const link = getByText('Next');

  fireEvent.click(link);

  expect(handlePageChange).toHaveBeenCalledTimes(1);
  expect(handlePageChange).toHaveBeenCalledWith(4);
});
