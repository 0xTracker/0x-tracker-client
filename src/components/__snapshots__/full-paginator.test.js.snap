// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render full size paginator 1`] = `
<div
  class="full-paginator__StyledPaginator-sc-4xhne0-1 hjnqpe"
>
  <p
    class="paging-summary___StyledP-sc-1a329kw-0 jpzpTu full-paginator___StyledPagingSummary-sc-4xhne0-2 joXbQg"
  >
    Displaying 
    1
     to 
    15
     of
     
    147
  </p>
  <div
    class="full-paginator__Pages-sc-4xhne0-0 hpCtnx"
  >
    <ul
      class="pagination"
    >
      <li
        class="page-item disabled"
      >
        <a
          aria-disabled="true"
          class="page-link"
          role="button"
          tabindex="0"
        >
          Previous
        </a>
      </li>
      <li
        class="page-item active"
      >
        <a
          aria-current="page"
          aria-label="Page 1 is your current page"
          class="page-link"
          role="button"
          tabindex="0"
        >
          1
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Page 2"
          class="page-link"
          role="button"
          tabindex="0"
        >
          2
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Page 3"
          class="page-link"
          role="button"
          tabindex="0"
        >
          3
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Page 4"
          class="page-link"
          role="button"
          tabindex="0"
        >
          4
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Page 5"
          class="page-link"
          role="button"
          tabindex="0"
        >
          5
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Page 6"
          class="page-link"
          role="button"
          tabindex="0"
        >
          6
        </a>
      </li>
      <li
        class="d-none"
      >
        <a
          role="button"
          tabindex="0"
        >
          ...
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-disabled="false"
          class="page-link"
          role="button"
          tabindex="0"
        >
          Next
        </a>
      </li>
    </ul>
  </div>
</div>
`;
